### ISSUE TITLE
PROTOCOL_ERROR - Thread 'main' panicked at 'called `Result::unwrap()`

### ISSUE THREAD

**Issue by icherukuri** [ducaale/ht#27]
Hello,

1. I tried a simple GET on https:// google.com. Internally ht got panicked and threw the below error.

```
➜  ~ ht get https://google.com
HTTP/2.0 400 Bad Request
content-length: 1555
content-type: text/html; charset=UTF-8
date: Sun, 07 Feb 2021 01:19:14 GMT
referrer-policy: no-referrer

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: reqwest::Error { kind: Body, source: hyper::Error(Body, Error { kind: Proto(PROTOCOL_ERROR) }) }', src/printer.rs:208:79
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

2. When tried with just http://google.com ht errored out with **too many redirects**

```
➜  ~ ht get http://google.com
Error: reqwest::Error { kind: Redirect, url: Url { scheme: "http", host: Some(Domain("www.google.com")), port: None, path: "/", query: None, fragment: None }, source: TooManyRedirects }
``` 

Environment
MacOS - 10.15

**Comment by payasrelekar** [comment]
When I tested yesterday, it was fine. But today someone reported while reviewing submission into nixpkgs and I can reproduce as well:

$ RUST_BACKTRACE=full ht-rs -v https://google.com
GET / HTTP/1.1
accept: */*
accept-encoding: gzip, deflate
connection: keep-alive
host: google.com

HTTP/2.0 400 Bad Request
content-length: 1555
content-type: text/html; charset=UTF-8
date: Sun, 07 Feb 2021 04:12:41 GMT
referrer-policy: no-referrer

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: reqwest::Error { kind: Body, source: hyper::Error(Body, Error { kind: Proto(PROTOCOL_ERROR) }) }', src/printer.rs:183:79
stack backtrace:
   0:     0x56097287442c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::haee811f030343ff1
   1:     0x56097289667f - core::fmt::write::h7db28da9f3cc3c42
   2:     0x56097286d275 - std::io::Write::write_fmt::h1b25d758ec27b86f
   3:     0x56097285c345 - std::panicking::default_hook::{{closure}}::h3cbc53da4ae5ec55
   4:     0x56097285c043 - std::panicking::default_hook::h6f6f96fba09cad38
   5:     0x56097285c99f - std::panicking::rust_panic_with_hook::h2b77a9d9f31f50a0
   6:     0x5609728748e7 - std::panicking::begin_panic_handler::{{closure}}::he7089cab3285baf3
   7:     0x56097287456c - std::sys_common::backtrace::__rust_end_short_backtrace::h5b2b5f10870a3b6b
   8:     0x56097285c542 - rust_begin_unwind
   9:     0x5609728942f1 - core::panicking::panic_fmt::h010f74ce205c2e26
  10:     0x56097289e803 - core::result::unwrap_failed::h21666bbf7d40cd85
  11:     0x5609724d49b1 - ht::main::{{closure}}::h5db0155593de3d08
  12:     0x5609724ba4ec - std::thread::local::LocalKey<T>::with::hae88770be0de4345
  13:     0x5609724e827d - tokio::runtime::enter::Enter::block_on::ha1086251b36d92c2
  14:     0x5609724fbe44 - tokio::runtime::thread_pool::ThreadPool::block_on::h088f434f4e2914b8
  15:     0x5609724b1da0 - tokio::runtime::Runtime::block_on::h0f2529a0c8adfc4f
  16:     0x5609724e8f68 - ht::main::h639aa9141f4b158f
  17:     0x5609724f6ad3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h3d7a95c43abb9a54
  18:     0x5609724f70dd - std::rt::lang_start::{{closure}}::h0b1fdeb2f8ed8ae7
  19:     0x56097286129b - std::rt::lang_start_internal::h5ef8963efbb238fe
  20:     0x5609724e9032 - main
  21:     0x7f0f5e129ded - __libc_start_main
  22:     0x5609724a71fa - _start
  23:                0x0 - <unknown>

**Comment by ducaale** [comment]
This is now fixed with https://github.com/ducaale/ht/pull/26 which is included in the ht v0.5.0 release.
```sh
$ get https://google.com
HTTP/2.0 301 Moved Permanently
alt-svc: h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
cache-control: private, max-age=2592000
content-length: 220
content-type: text/html; charset=UTF-8
date: Sun, 07 Feb 2021 08:42:25 GMT
expires: Sun, 07 Feb 2021 08:42:25 GMT
location: https://www.google.com/
p3p: CP="This is not a P3P policy! See g.co/p3phelp for more info."
server: gws
set-cookie: CONSENT=PENDING+523; expires=Fri, 01-Jan-2038 00:00:00 GMT; path=/; domain=.google.com
x-frame-options: SAMEORIGIN
x-xss-protection: 0

<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>
```
```sh
$ get http://google.com
HTTP/1.1 301 Moved Permanently
cache-control: public, max-age=2592000
content-length: 219
content-type: text/html; charset=UTF-8
date: Sun, 07 Feb 2021 08:44:11 GMT
expires: Tue, 09 Mar 2021 08:44:11 GMT
location: http://www.google.com/
server: gws
x-frame-options: SAMEORIGIN
x-xss-protection: 0

<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
```

Feel free to reopen the issue if you encounter related errors, Thanks.
